File: 2022/typescript/day1.ts
import { readFileSync } from 'fs';

const input = readFileSync('../input/day1.txt', 'utf-8');
const input_arr = input.split('\r\n').map(x => { return parseInt(x, 10) })

const elfs: number[][] = [];
let currentElf: number[] = [];
let elfsums: number[] = [];

input_arr.forEach(line => {
    if (isNaN(line)) {
        elfs.push(currentElf)
        currentElf = [];
    } else {
        currentElf.push(line)
    }
});

elfs.push(currentElf)
currentElf = []


elfs.forEach(elf => {
    let calorieCount = 0;
    elf.forEach(num => {
        calorieCount += num;
    })
    elfsums.push(calorieCount)
})

const elfsums_sorted = elfsums.sort((a, b) => b - a)

// part 1
console.log(elfsums_sorted[0]);

// part 2
console.log(elfsums_sorted.slice(0, 3).reduce((a, b) => a + b));

File: python/2022/day1.py
with open("../../input/2022/day1.txt") as f:
    input_arr = f.read().split('\n')

current_elf = []
elfs = []
elf_sums = []

for row in input_arr:
    if row == '':
        elfs.append(current_elf)
        current_elf = []
    else:
        row = int(row)
        current_elf.append(row)

elfs.append(current_elf)
current_elf = []

for elf in elfs:
    elf_sums.append(sum(elf))
sorted_elf_sums = sorted(elf_sums, reverse=True)

# Part 1
print(sorted_elf_sums[0])

# Part 2
print(sorted_elf_sums[0] + sorted_elf_sums[1] + sorted_elf_sums[2])
